### PR TITLE
Table with EOL dates for all versions of all libraries

### LINKED ISSUE #49: List EOL dates for all versions of all libraries
We have EOL dates for collections / distributions listed [here](https://ignitionrobotics.org/docs/all/releases), but from just that list, it's complicated to tell the EOL date for each version of each library.

For new releases, we have the implicit rule that the libraries EOL together with the last collection that uses that version. For example, `ign-gazebo1` EOL'ed with Acropolis, but `ign-math6` doesn't have a EOL date yet, because it may be used by Ign-E.

But for versions that predate distributions, we've never specified their EOL dates, so we need to take other things into consideration, such as:

* Is a supported Gazebo version using them?
* Are they released as official Debian packages into supported distros? (i.e. `ign-math4` is an [official package](https://packages.ubuntu.com/source/focal/ignition-math4) in Ubuntu Focal)

I'm putting together a spreadsheet compiling this information, and my plan is to publish it in a machine-readable format that can be used to display on the documentation site, and also to be queried by other systems. I can imagine it will live either in https://github.com/ignition-tooling/release-tools or https://github.com/ignition-tooling/gazebodistro.

CC @j-rivero @scpeters

### PR DESCRIPTION
**Pull request by chapulina**
# 🎉 New feature

Closes #49

## Summary

Moving the table from an internal spreadsheet.

@j-rivero , do the package columns look good to you?

## Checklist
- [x] Signed all commits for DCO
- [x] Updated documentation (as needed)
- [x] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge**

🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸

### CODE REVIEW

File: tools/versions.md [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+How to read the columns:
+
+* **Package**: Each library or application that is supported by the Open Robotics simulation team.
+* **Version**: Major version for each package.
+* **Status**
+    * **EOL**: Reached End of Life and is no longer supported.
+    * **stable**: Release is stable and may get bug fixes and backwards-compatible features.
+    * **unstable**: No stable release has been made for that major version yet.
+* **Packages**: Whether debian packages for a version are provided by specific repositories. Each letter refers to an Ubuntu release (i.e. T for Trusty, B for Bionic...). The repositories are the following:
+    * **Ubuntu**: https://packages.ubuntu.com/
```
Comment (j-rivero) [inline tools/versions.md:14]:
```suggestion
    * **Ubuntu**: https://packages.ubuntu.com/ (official Ubuntu archive packages)
```

File: tools/versions.md [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+    * **OSRF**: https://packages.osrfoundation.org/
```
Comment (j-rivero) [inline tools/versions.md:15]:
```suggestion
    * **OSRF**: https://packages.osrfoundation.org/ (offical repository of the Ignition/Gazebo project)
```

File: tools/versions.md [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+    * **ROS**: https://packages.ros.org/
```
Comment (j-rivero) [inline tools/versions.md:16]:
```suggestion
    * **ROS**: https://packages.ros.org/ (package imported from OSRF repository into ROS)
```

**Review by j-rivero — COMMENTED** [review]
The table is great! made some minor comments, I've included the new version of Ubuntu under development. Only miss the Debian repositories, we probably have less users than in Ubuntu and the table is pretty dense.

File: tools/versions.md [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+Summary of versions for all Ignition libraries and their support dates.
+
+How to read the columns:
+
+* **Package**: Each library or application that is supported by the Open Robotics simulation team.
+* **Version**: Major version for each package.
+* **Status**
+    * **EOL**: Reached End of Life and is no longer supported.
+    * **stable**: Release is stable and may get bug fixes and backwards-compatible features.
+    * **unstable**: No stable release has been made for that major version yet.
```
[the remark this answers is not in the document]
Comment (chapulina) [reply tools/versions.md]: Good point, I went with `note-released` because some of them aren't under active development.

**Review by j-rivero — APPROVED** [review]

**Pull request merged**